devito/function.py

```diff
@@ -108,6 +109,15 @@ class Function(TensorFunction):
        :class:`Function` objects are assumed to be constant in time
        and therefore do not support time derivatives. Use
        :class:`TimeFunction` for time-varying grid data.
+
+    .. note::
+
+       The tuple :param staggered: contains a ``1`` in each dimension
+       entry that should be staggered, and ``0`` otherwise. For example,
+       ``staggered=(1, 0, 0)`` entails discretization on horizontal edges,
+       ``staggered=(0, 0, 1)`` entails discretization on vertical edges,
+       ``staggered=(0, 1, 1)`` entails discretization side facets and
+       ``staggered=(1, 1, 1)`` entails discretization on cells.
     """
```

Comment:
CAn we name these (as a Dict or something) to make it more user friendly?

Reply:
Also how does this extend to 2D?

Reply:
For 2D, simply supply a 2-tuple for `Function` and a 3-tuple for `TimeFunction` (stagger in time). I thought about the naming conventions, and didn't find any obvious ones. But in a future PR we should definitely add this. This PR is pretty much bare-metal to enable experimentation.